kolena/workflow/annotation.py

```diff
@@ -328,6 +330,53 @@ class ScoredLabel(Label):
 """Alias for [`ScoredLabel`][kolena.workflow.annotation.ScoredLabel]."""
 
 
+@dataclass(frozen=True, config=ValidatorConfig)
+class TimeSegment(Annotation):
+    """
+    Segment of time in the associated audio or video file.
+
+    When a `group` is specified, segments are displayed on Kolena with different colors for each group present in a
+    `List[TimeSegment]`.
```

Comment:
Looking for feedback on this. Rather than adding `group` as an optional field to the annotation directly, is it sufficient to document this special behavior here?

Reply:
Where should I add `group` to? Is it like

```
class MyGT(GroundTruth):
   group: List[TimeSegment]

mygt = MyGT(groupd=[TimeSegment(..), TimeSegment(...)])
```

Reply:
You'd add it to each TimeSegment instance. e.g.

```python
class GT(GroundTruth):
   transcription: List[TimeSegment]

gt = GT(transcription=[
    TimeSegment(start=0, end=5, group="speakerA", label="Hello and welcome to our guest XYZ"),
    TimeSegment(start=6, end=10, group="speakerB", label="Thank you for having me!"),
    TimeSegment(start=11, end=20, group="speakerA", label="XYZ is joining us from [...]"),
    ...
])
```

Reply:
what's reason not to add it as an optional field? It should be omitted from persistence if `None`.

Reply:
Until [3.10](https://www.trueblade.com/blogs/news/python-3-10-new-dataclass-features), dataclasses could not be extended (here to `LabeledTimeSegment`, `ScoredTimeSegment`, `ScoredLabeledTimeSegment`) with required fields if the base class had optional fields.

Reply:
Forgot about that. I think maybe a code snippet could help.

Reply:
Added example to docstring 👍